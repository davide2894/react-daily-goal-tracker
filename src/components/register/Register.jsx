import React, { useState } from "react";
import { registerWithEmailAndPassword } from "../../firebase";

function Register() {
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  function handleSubmit(evt) {
    evt.preventDefault();
    registerWithEmailAndPassword(name, email, password);
  }

  return (
    <div
      data-testid="registerComponentTestElement"
      className="myAccount__form myAccount__form--register">
      <p>First time here? Register an account to start setting goals! </p>
      <br></br>
      <form onSubmit={handleSubmit}>
        <div className="form-control">
          <label>Name</label>
          <input
            type="text"
            name="name"
            placeholder="Enter name"
            value={name}
            onChange={(evt) => setName(evt.target.value)}
          />
        </div>
        <div className="form-control">
          <label>Email</label>
          <input
            type="text"
            name="email"
            placeholer="Enter email"
            value={email}
            onChange={(evt) => setEmail(evt.target.value)}
          />
        </div>
        <div className="form-control">
          <label>Password</label>
          <input
            type="password"
            name="password"
            placeholder="Enter password"
            value={password}
            onChange={(evt) => setPassword(evt.target.value)}
          />
        </div>
        <div className="form-control">
          <label></label>
          <button type="submit">Register</button>
        </div>
      </form>
    </div>
  );
}

export default Register;
