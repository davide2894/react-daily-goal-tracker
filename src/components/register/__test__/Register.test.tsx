import Register from "../Register";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("Register component renders correctly", () => {
  render(<Register />);
  const registerComponentWrapper = screen.getByTestId(
    "registerComponentTestElement"
  );
  expect(registerComponentWrapper).toBeInTheDocument();
  expect(registerComponentWrapper).toHaveClass("myAccount__form ");
  expect(registerComponentWrapper).toHaveClass("myAccount__form--register");
});

test("Register component matches its own snapshot", () => {
  const tree = renderer.create(<Register />).toJSON();
  expect(tree).toMatchSnapshot();
});
