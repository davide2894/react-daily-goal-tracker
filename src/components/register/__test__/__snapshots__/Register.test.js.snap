// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component matches its own snapshot 1`] = `
<div
  className="myAccount__form myAccount__form--register"
  data-testid="registerComponentTestElement"
>
  <p>
    First time here? Register an account to start setting goals! 
  </p>
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-control"
    >
      <label>
        Name
      </label>
      <input
        name="name"
        onChange={[Function]}
        placeholder="Enter name"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label>
        Email
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholer="Enter email"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label>
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label />
      <button
        type="submit"
      >
        Register
      </button>
    </div>
  </form>
</div>
`;
